repair/repair.cc

```diff
@@ -568,6 +582,7 @@ repair_info::repair_info(repair_service& repair,
     , reason(reason_)
     , total_rf(db.local().find_keyspace(keyspace).get_effective_replication_map()->get_replication_factor())
     , nr_ranges_total(ranges.size())
+    , as(asp)
     , _hints_batchlog_flushed(std::move(hints_batchlog_flushed)) {
     if (as != nullptr) {
         _abort_subscription = as->subscribe([this] () noexcept { abort(); });
```

Comment:
Do we still need `_abort_subscription`and `aborted` flag in `repair_info`?
We could just check `as->abort_requested()`, since abort_source pointer is kept in repair_info.

Reply:
This is how the abort trigger is pased to repair_info. Otherwise, the ops info layer could call the abort method directly, if it would keep a foreign shared prr to the repair info (like children tasks) but that's a bigger change

Reply:
the repair info abort source is optional. I'm not sure if we can grt rid of the abort member

Reply:
@Deexie just to make it clear that after converting to the task manager infrastructure, all tasks should be abortable internally, those that are nonabortable are so just from the api perspective.
We can keep an abort_source at the generic task impl and request abort via the task class.
Then, the task impl replacing repair info can use it as you suggested above.

With that we'll be able to abort individual tasks, like a particular node op, by traversing the tree and aborting the descendents individually.

When a global abort is requested, e.g. on shutdown, we subscribe to at the module level, which should abort all tasks it owns by using the task map.  Similar to the meta layer in storage service in this series.